modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -668,39 +739,20 @@ public void writeByteBuffer(ByteBuffer val) {
     /** {@inheritDoc} */
     @Override
     public void writeUuid(UUID val) {
-        switch (uuidState) {
-            case 0:
-                writeBoolean(val == null);
-
-                if (!lastFinished || val == null) {
-                    return;
-                }
-
-                uuidState++;
-
-                //noinspection fallthrough
-            case 1:
-                writeLong(val.getMostSignificantBits());
-
-                if (!lastFinished) {
-                    return;
-                }
-
-                uuidState++;
-
-                //noinspection fallthrough
-            case 2:
-                writeLong(val.getLeastSignificantBits());
+        if (val == null) {
+            writeBoolean(true);
+        } else {
+            lastFinished = remainingInternal() >= 1 + 2 * Long.BYTES;
 
-                if (!lastFinished) {
-                    return;
-                }
+            if (lastFinished) {
+                int pos = buf.position();
 
-                uuidState = 0;
-                break;
+                GridUnsafe.putBoolean(heapArr, baseOff + pos, false);
+                GridUnsafe.putLongLittleEndian(heapArr, baseOff + pos + 1, val.getMostSignificantBits());
+                GridUnsafe.putLongLittleEndian(heapArr, baseOff + pos + 9, val.getLeastSignificantBits());
 
-            default:
-                throw new IllegalArgumentException("Unknown uuidState: " + uuidState);
+                setPosition(pos + 17);
```

Comment:
There are quite a lot of magic numbers in this class

At least let's change calculated 17 with 1 + 8 + 8?

Reply:
Sure